pkg/services/ngalert/api/api_convert_prometheus.go

```diff
@@ -105,10 +158,17 @@ func (srv *ConvertPrometheusSrv) RouteConvertPrometheusGetRules(c *contextmodel.
 func (srv *ConvertPrometheusSrv) RouteConvertPrometheusDeleteNamespace(c *contextmodel.ReqContext, namespaceTitle string) response.Response {
 	logger := srv.logger.FromContext(c.Req.Context())
 
-	logger.Debug("Looking up folder in the root by title", "folder_title", namespaceTitle)
-	namespace, err := srv.ruleStore.GetNamespaceInRootByTitle(c.Req.Context(), namespaceTitle, c.SignedInUser.GetOrgID(), c.SignedInUser)
+	workingFolderUID := getWorkingFolderUID(c)
+	logger = logger.New("working_folder_uid", workingFolderUID)
+
+	logger.Debug("Looking up folder by title", "folder_title", namespaceTitle)
+	namespace, err := srv.ruleStore.GetNamespaceByTitle(c.Req.Context(), namespaceTitle, c.SignedInUser.GetOrgID(), c.SignedInUser, workingFolderUID)
+	if errors.Is(err, dashboards.ErrFolderNotFound) {
+		// if the folder does not exist, return 202 Accepted
```

Comment:
mimir returns 404 when deleting a non-existing namespace (and 202 when it existed and has been deleted):

```
curl -i -X DELETE http://127.0.0.1:9090/prometheus/config/v1/rules/unknown_namespace
HTTP/1.1 404 Not Found
```

but 202 when deleting a non-existing group or trying to delete a group from a non-existing namespace:

```
curl -i -X DELETE http://127.0.0.1:9090/prometheus/config/v1/rules/unknown_namespace/group_123
HTTP/1.1 202 Accepted
...
```

I changed all these to 202 in our API instead of keeping 404 and 202 since it looks more consistent and doesn't matter for the mimirtool.

Reply:
hmm, this is weird. I tried with both cloud Mimir instances and I see the same 202 in case of a non-existing group. I'll follow up and check what's going on, but I will change it here to 404 as this seems to be the intended behaviour according to the Mimir code.

Reply:
Updated: https://github.com/grafana/grafana/pull/101406/commits/4e1b56c326f80c800a278aedbe36363611590b0d

Reply:
I did a small investigation and this is an AWS S3 thing: it always returns 204 even if the object does not exist. Azure returns 404 though, so both should be fine :)